docs/developers/local-rules.md

```diff
@@ -0,0 +1,212 @@
+---
+id: local-rules
+title: Local rules
+---
+
+If your custom rules only target a single codebase or repository, you can
+define and develop these rules as part of the same sbt build. This way, you do
+not have to version your rules and publish them to a Maven or Ivy repository.
+
+When implementing rules that are very specific to your domain, this strategy
+can be interesting as it allows rules to be added or extended in the same
+change sets as the domain sources they will be run against.
+
+## Prerequisite
+
+Make sure the sbt plugin, as well as the Scala compiler plugin and options [are
+set up correctly](../users/installation.md#sbt).
+
+Although it is possible to define your rules in a Scala binary version that
+does not match your build, it is highly recommended that you align them via:
+
+```diff
+ // build.sbt
++scalafixScalaBinaryVersion in ThisBuild :=
++  CrossVersion.binaryScalaVersion(scalaVersion.value)
+```
+
+> Note that any potential external Scalafix rule, loaded with the
+> `scalafixDependencies` setting key, must be built and published against the
+> same Scala binary version.
+
+## Single-project sbt build
+
+If you do not have any sub-project in your build, custom rules should be
+defined under the Scalafix Ivy configuration sources directory, which by
```

Comment:
we found this to be a problem with IntelliJ's built-in compiler, as unknown Ivy configurations fallback to `Compile` without any proper eviction. I'll follow up with some notes & recommendations when that's a problem.